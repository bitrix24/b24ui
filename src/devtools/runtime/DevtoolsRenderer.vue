<script setup lang="ts">
import { onUnmounted, onMounted, reactive } from 'vue'
import { pascalCase } from 'scule'
import { defineAsyncComponent, useColorMode, useRoute } from '#imports'

const route = useRoute()
const component = route.query?.example
  ? defineAsyncComponent(() => import(`./examples/${route.query.example}.vue`))
  : route.params?.slug && defineAsyncComponent(() => import(`../../runtime/components/${pascalCase(route.params.slug as string)}.vue`))

const state = reactive<{ slots?: any, props?: any }>({})

function onUpdateRenderer(event: Event & { data?: any }) {
  state.props = { ...event.data.props }
  state.slots = { ...event.data.slots }
}

const colorMode = useColorMode()
function setColorMode(event: Event & { isDark?: boolean }) {
  colorMode.preference = event.isDark ? 'dark' : 'light'
}

onMounted(() => {
  window.parent.addEventListener('bitrix24-ui-devtools:update-renderer', onUpdateRenderer)
  window.parent.addEventListener('bitrix24-ui-devtools:set-color-mode', setColorMode)
})

onUnmounted(() => {
  window.parent.removeEventListener('bitrix24-ui-devtools:update-renderer', onUpdateRenderer)
  window.parent.removeEventListener('bitrix24-ui-devtools:set-color-mode', setColorMode)
})

onMounted(async () => {
  const event: Event = new Event('bitrix24-ui-devtools:component-loaded')
  window.parent.dispatchEvent(event)
})

onMounted(() => {
  if (!route.query?.example) return
})
</script>

<template>
  <div id="b24ui-devtools-renderer" class="bitrix24-ui-component-renderer">
    <B24App :toaster="null">
      <component :is="component" v-if="component" v-bind="state.props" :class="state?.slots?.base" :ui="state.slots" />
    </B24App>
  </div>
</template>

<style>
.bitrix24-ui-component-renderer {
  position: 'relative';
  height: 100vh;
  width: 100vw;

  padding: 32px;

  display: flex;
  justify-content: center;
  align-items: center;

  background-image: url("data:image/svg+xml;charset=utf-8,%3Csvg xmlns='http://www.w3.org/2000/svg' width='20' height='20' transform='scale(3)'%3E%3Crect width='100%25' height='100%25' fill='%23fff'/%3E%3Cpath fill='none' stroke='hsla(0, 0%25, 98%25, 1)' stroke-width='.2' d='M10 0v20ZM0 10h20Z'/%3E%3C/svg%3E");
  background-size: 40px 40px;
}

.dark .bitrix24-ui-component-renderer {
background-image: url("data:image/svg+xml;charset=utf-8,%3Csvg xmlns='http://www.w3.org/2000/svg' width='20' height='20' transform='scale(3)'%3E%3Crect width='100%25' height='100%25' fill='hsl(0, 0%25, 8.5%25)'/%3E%3Cpath fill='none' stroke='hsl(0, 0%25, 11.0%25)' stroke-width='.2' d='M10 0v20ZM0 10h20Z'/%3E%3C/svg%3E");
  background-size: 40px 40px;
}
</style>
