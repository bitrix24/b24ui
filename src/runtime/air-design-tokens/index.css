@import './001_b24_global.css';
@import './003_b24_context-content-light.css';
@import './004_b24_context-content-dark.css';
@import './005_b24_context-edge-light.css';
@import './006_b24_context-edge-dark.css';
@import './007_b24_global.css';
@import './008_ui_global.css';
@import './009_b24_tools.css';
@import './components/scrollbar.css';
@import './components/navigation-menu.css';
@import './components/badge-counter.css';
