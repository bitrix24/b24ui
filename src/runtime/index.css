@plugin '@bitrix24/b24style';
@import '#build/b24ui.css';
@import './keyframes.css';
@import './air-design-tokens/index.css';

@layer base {
  body,
  .sidebar-layout.--inner {
    /* @see air-design-tokens/???_b24_context*.css */
    --air-theme-background:
      var(--air-theme-bg-image, none)
      var(--air-theme-bg-position, 0 0) / var(--air-theme-bg-size, auto)
      var(--air-theme-bg-repeat, repeat)
      var(--air-theme-bg-attachment, scroll)
      var(--air-theme-bg-color, transparent)
    ;

    --air-theme-background-blurred:
      var(--air-theme-bg-image-blurred, none)
      var(--air-theme-bg-position, 0 0) / var(--air-theme-bg-size, auto)
      var(--air-theme-bg-repeat, repeat)
      var(--air-theme-bg-attachment, scroll)
    ;
  }

  /* @see src/theme/sidebar-layout.ts */
  body {
    background: var(--air-theme-background);
  }

  .sidebar-layout.--inner {
    background: var(--air-theme-background);
  }

  body {
    overflow-y: scroll!important;
    padding-right: 0!important;
    scrollbar-gutter: stable;
  }

  .--app[data-state="loading"] {
    overflow: hidden;
  }
}
