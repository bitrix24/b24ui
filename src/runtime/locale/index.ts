/**
 * en English
 * de Deutsch
 * la Español
 *
 * br Português (Brasil)
 * fr Français
 * it Italiano
 *
 * pl Polski
 * ru Русский
 * ua Українська
 *
 * tr Türkçe
 * sc 中文（简体）
 * tc 中文（繁體)
 *
 * ja 日本語
 * vn Tiếng Việt
 * id Bahasa Indonesia
 *
 * ms Bahasa Melayu
 * th ภาษาไทย
 * ar عربي,
 *
 * kz Қазақша
 */
export { default as en } from './en'
export { default as de } from './de'
export { default as la } from './la'

export { default as br } from './br'
export { default as fr } from './fr'
export { default as it } from './it'

export { default as pl } from './pl'
export { default as ru } from './ru'
export { default as ua } from './ua'

export { default as tr } from './tr'
export { default as sc } from './sc'
export { default as tc } from './tc'

export { default as ja } from './ja'
export { default as vn } from './vn'
export { default as id } from './id'

export { default as ms } from './ms'
export { default as th } from './th'
export { default as ar } from './ar'

export { default as kz } from './kz'
