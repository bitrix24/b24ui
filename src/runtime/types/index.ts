export * from '../components/App.vue'
export * from '../components/Avatar.vue'
export * from '../components/AvatarGroup.vue'
export * from '../components/Button.vue'
export * from '../components/Checkbox.vue'
export * from '../components/Chip.vue'
export * from '../components/Container.vue'
export * from '../components/FormField.vue'
export * from '../components/Input.vue'
export * from '../components/Kbd.vue'
export * from '../components/Link.vue'
export * from '../components/Progress.vue'
export * from '../components/RadioGroup.vue'
export * from '../components/Separator.vue'
export * from '../components/Skeleton.vue'
export * from '../components/content/SkeletonTaskAdd.vue'
export * from '../components/Switch.vue'
export * from '../components/Textarea.vue'
export * from '../components/Tooltip.vue'
export * from './form'
export * from './locale'
