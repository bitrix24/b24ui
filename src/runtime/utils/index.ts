import { isEqual } from 'ohash/utils'
import type { GetItemKeys, NestedItem } from '../types/utils'

export function pick<Data extends object, Keys extends keyof Data>(data: Data, keys: Keys[]): Pick<Data, Keys> {
  const result = {} as Pick<Data, Keys>

  for (const key of keys) {
    result[key] = data[key]
  }

  return result
}

export function omit<Data extends object, Keys extends keyof Data>(data: Data, keys: Keys[]): Omit<Data, Keys> {
  const result = { ...data }

  for (const key of keys) {
    // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
    delete result[key]
  }

  return result as Omit<Data, Keys>
}

export function get(object: Record<string, any> | undefined, path: (string | number)[] | string, defaultValue?: any): any {
  if (typeof path === 'string') {
    path = path.split('.').map((key) => {
      const numKey = Number(key)
      return Number.isNaN(numKey) ? key : numKey
    })
  }

  let result: any = object

  for (const key of path) {
    if (result === undefined || result === null) {
      return defaultValue
    }

    result = result[key]
  }

  return result !== undefined ? result : defaultValue
}

export function set(object: Record<string, any>, path: (string | number)[] | string, value: any): void {
  if (typeof path === 'string') {
    path = path.split('.').map((key) => {
      const numKey = Number(key)
      return Number.isNaN(numKey) ? key : numKey
    })
  }

  path.reduce((acc, key, i) => {
    if (acc[key] === undefined) acc[key] = {}
    if (i === path.length - 1) acc[key] = value
    return acc[key]
  }, object)
}

export function looseToNumber(val: any): any {
  const n = Number.parseFloat(val)
  return Number.isNaN(n) ? val : n
}

export function compare<T>(value?: T, currentValue?: T, comparator?: string | ((a: T, b: T) => boolean)) {
  if (value === undefined || currentValue === undefined) {
    return false
  }

  if (typeof value === 'string') {
    return value === currentValue
  }

  if (typeof comparator === 'function') {
    return comparator(value, currentValue)
  }

  if (typeof comparator === 'string') {
    return get(value!, comparator) === get(currentValue!, comparator)
  }

  return isEqual(value, currentValue)
}

export function isEmpty(value: unknown): boolean {
  if (value == null) {
    return true
  }

  if (typeof value === 'boolean' || typeof value === 'number') {
    return false
  }

  if (typeof value === 'string') {
    return value.trim().length === 0
  }

  if (Array.isArray(value)) {
    return value.length === 0
  }

  if (value instanceof Map || value instanceof Set) {
    return value.size === 0
  }

  if (value instanceof Date || value instanceof RegExp || typeof value === 'function') {
    return false
  }

  if (typeof value === 'object') {
    for (const _ in value as object) {
      if (Object.prototype.hasOwnProperty.call(value, _)) {
        return false
      }
    }
    return true
  }

  return false
}

export function getDisplayValue<T, V>(
  items: T[],
  value: V | undefined | null,
  options: {
    valueKey?: GetItemKeys<T>
    labelKey?: keyof NestedItem<T>
  } = {}
): string | undefined {
  const { valueKey, labelKey } = options

  if (isEmpty(value)) {
    return undefined
  }

  const foundItem = items.find((item) => {
    const itemValue = (typeof item === 'object' && item !== null && valueKey)
      ? get(item, valueKey as string)
      : item
    return compare(itemValue, value)
  })

  const source = foundItem ?? value

  if (source === null || source === undefined) {
    return undefined
  }

  if (typeof source === 'object') {
    return labelKey ? get(source as Record<string, any>, labelKey as string) : undefined
  }

  return String(source)
}

export function isArrayOfArray<A>(item: A[] | A[][]): item is A[][] {
  return Array.isArray(item[0])
}

export function mergeClasses(appConfigClass?: string | string[], propClass?: string) {
  if (!appConfigClass && !propClass) {
    return ''
  }

  return [
    ...(Array.isArray(appConfigClass) ? appConfigClass : [appConfigClass]),
    propClass
  ].filter(Boolean)
}

export function getSlotChildrenText(children: any) {
  return children.map((node: any) => {
    if (!node.children || typeof node.children === 'string') return node.children || ''
    else if (Array.isArray(node.children)) return getSlotChildrenText(node.children)
    else if (node.children.default) return getSlotChildrenText(node.children.default())
  }).join('')
}

export function transformUI(ui: any, uiProp?: any) {
  return Object.entries(ui).reduce((acc, [key, value]) => {
    acc[key] = typeof value === 'function' ? value({ class: uiProp?.[key] }) : value
    return acc
  }, uiProp || {})
}

export * from './content'
