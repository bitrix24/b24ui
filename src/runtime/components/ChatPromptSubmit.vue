<script lang="ts">
import type { ChatStatus } from 'ai'
import type { AppConfig } from '@nuxt/schema'
import theme from '#build/b24ui/chat-prompt-submit'
import type { ButtonProps, ButtonSlots, IconComponent } from '../types'
import type { ComponentConfig } from '../types/tv'

type ChatPromptSubmit = ComponentConfig<typeof theme, AppConfig, 'chatPromptSubmit'>

export interface ChatPromptSubmitProps extends /** @vue-ignore */ Pick<ButtonProps, 'size' | 'label'> {
  status?: ChatStatus
  /**
   * The icon displayed in the button when the status is `ready`.
   * @defaultValue icons.arrowUp
   * @IconComponent
   */
  icon?: IconComponent
  /**
   * The color of the button when the status is `ready`.
   * @defaultValue 'air-primary'
   */
  color?: ButtonProps['color']
  /**
   * The icon displayed in the button when the status is `streaming`.
   * @defaultValue icons.stop
   * @IconComponent
   */
  streamingIcon?: IconComponent
  /**
   * The color of the button when the status is `streaming`.
   * @defaultValue 'air-secondary-no-accent'
   */
  streamingColor?: ButtonProps['color']
  /**
   * The icon displayed in the button when the status is `submitted`.
   * @defaultValue icons.stop
   * @IconComponent
   */
  submittedIcon?: IconComponent
  /**
   * The color of the button when the status is `submitted`.
   * @defaultValue 'air-secondary-no-accent'
   */
  submittedColor?: ButtonProps['color']
  /**
   * The icon displayed in the button when the status is `error`.
   * @defaultValue icons.reload
   * @IconComponent
   */
  errorIcon?: IconComponent
  /**
   * The color of the button when the status is `error`.
   * @defaultValue 'air-primary-alert'
   */
  errorColor?: ButtonProps['color']
  b24ui?: ChatPromptSubmit['slots'] & ButtonProps['b24ui']
  class?: any
}

export interface ChatPromptSubmitEmits {
  stop: []
  reload: []
}
</script>

<script setup lang="ts">
import { computed } from 'vue'
import { useAppConfig } from '#imports'
import { useLocale } from '../composables/useLocale'
import { transformUI } from '../utils'
import { tv } from '../utils/tv'
import icons from '../dictionary/icons'
import B24Button from './Button.vue'

const props = withDefaults(defineProps<ChatPromptSubmitProps>(), {
  status: 'ready',
  streamingColor: 'air-secondary-no-accent',
  submittedColor: 'air-secondary-no-accent',
  errorColor: 'air-primary-alert'
})
const emits = defineEmits<ChatPromptSubmitEmits>()
const slots = defineSlots<ButtonSlots>()

const { t } = useLocale()
const appConfig = useAppConfig() as ChatPromptSubmit['AppConfig']

const buttonProps = computed(() => ({
  ready: {
    icon: props.icon || icons.arrowUp,
    color: props.color,
    type: 'submit' as const
  },
  submitted: {
    icon: props.submittedIcon || icons.stop,
    color: props.submittedColor,
    onClick() {
      emits('stop')
    }
  },
  streaming: {
    icon: props.streamingIcon || icons.stop,
    color: props.streamingColor,
    onClick() {
      emits('stop')
    }
  },
  error: {
    icon: props.errorIcon || icons.reload,
    color: props.errorColor,
    onClick() {
      emits('reload')
    }
  }
})[props.status])

// eslint-disable-next-line vue/no-dupe-keys
const b24ui = computed(() => tv({ extend: tv(theme), ...(appConfig.b24ui?.chatPromptSubmit || {}) })())
</script>

<template>
  <B24Button
    :aria-label="t('chatPromptSubmit.label')"
    v-bind="buttonProps"
    :class="b24ui.base({ class: [props.b24ui?.base, props.class] })"
    :b24ui="transformUI(b24ui, props.b24ui)"
  >
    <template v-for="(_, name) in slots" #[name]="slotData">
      <slot :name="name" v-bind="slotData" />
    </template>
  </B24Button>
</template>
