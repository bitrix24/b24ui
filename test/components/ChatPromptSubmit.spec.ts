import { describe, it, expect } from 'vitest'
import { axe } from 'vitest-axe'
import { mountSuspended } from '@nuxt/test-utils/runtime'
import ChatPromptSubmit from '../../src/runtime/components/ChatPromptSubmit.vue'
import type { ChatPromptSubmitProps } from '../../src/runtime/components/ChatPromptSubmit.vue'
import ComponentRender from '../component-render'
import Search2Icon from '@bitrix24/b24icons-vue/main/Search2Icon'

describe('ChatPromptSubmit', () => {
  const statuses = ['ready', 'submitted', 'streaming', 'error'] as any

  it.each([
    // Props
    ['with icon', { props: { icon: Search2Icon } }],
    ...statuses.map((status: string) => [`with status ${status}`, { props: { status } }]),
    ['with class', { props: { class: '' } }]
  ])('renders %s correctly', async (nameOrHtml: string, options: { props?: ChatPromptSubmitProps }) => {
    const html = await ComponentRender(nameOrHtml, options, ChatPromptSubmit)
    expect(html).toMatchSnapshot()
  })

  it('passes accessibility tests', async () => {
    const wrapper = await mountSuspended(ChatPromptSubmit, {
      props: {
        status: 'ready'
      }
    })

    expect(await axe(wrapper.element)).toHaveNoViolations()
  })
})
