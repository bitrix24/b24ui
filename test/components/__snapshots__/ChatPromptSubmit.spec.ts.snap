// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChatPromptSubmit > renders with class correctly 1`] = `
"<button type="submit" aria-label="Send" class="ui-btn font-[family-name:var(--ui-font-family-primary)] select-none cursor-pointer inline-flex items-center relative outline-transparent focus-visible:outline-2 focus-visible:outline-offset-2 disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-30 aria-disabled:opacity-30 transition duration-0 ease-linear border-(length:--ui-btn-border-width) text-(--ui-btn-color) bg-(--ui-btn-background) border-(--ui-btn-border-color) hover:text-(--ui-btn-color-hover) hover:bg-(--ui-btn-background-hover) hover:border-(--ui-btn-border-color-hover) focus:text-(--ui-btn-color-hover) focus:bg-(--ui-btn-background-hover) focus:border-(--ui-btn-border-color-hover) active:text-(--ui-btn-color-active) active:bg-(--ui-btn-background-active) active:border-(--ui-btn-border-color-active) disabled:bg-(--ui-btn-background) disabled:border-(--ui-btn-border-color) aria-disabled:bg-(--ui-btn-background) aria-disabled:border-(--ui-btn-border-color) focus-visible:outline-(--ui-btn-background) ring-(--ui-btn-background-hover) focus:outline-none focus-visible:ring-(--ui-btn-background-hover) h-(--ui-btn-height) text-(length:--ui-btn-font-size) leading-(--ui-btn-height) font-(--ui-btn-font-weight) --style-filled ui-btn-md normal-case --air w-(--ui-btn-height) rounded-full">
  <!--v-if-->
  <div class="w-full inline-flex items-center justify-center gap-(--ui-btn-icon-space) ps-[4px] pe-[4px]"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" aria-hidden="true" data-slot="icon" class="text-(--ui-btn-color) shrink-0 size-(--ui-btn-icon-size)">
      <path fill="currentColor" d="m7.986 11.103 6.697.618a.247.247 0 0 1-.002.492l-6.686.56a1 1 0 0 0-.903.834l-.732 4.418a1 1 0 0 0 1.448 1.05l11.56-6.016a1 1 0 0 0 .019-1.764L7.84 4.988a1 1 0 0 0-1.466 1.044l.716 4.242a1 1 0 0 0 .895.829"></path>
    </svg>
    <!--v-if-->
    <!--v-if-->
  </div>
</button>"
`;

exports[`ChatPromptSubmit > renders with icon correctly 1`] = `
"<button type="submit" aria-label="Send" class="ui-btn font-[family-name:var(--ui-font-family-primary)] select-none cursor-pointer inline-flex items-center relative outline-transparent focus-visible:outline-2 focus-visible:outline-offset-2 disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-30 aria-disabled:opacity-30 transition duration-0 ease-linear border-(length:--ui-btn-border-width) text-(--ui-btn-color) bg-(--ui-btn-background) border-(--ui-btn-border-color) hover:text-(--ui-btn-color-hover) hover:bg-(--ui-btn-background-hover) hover:border-(--ui-btn-border-color-hover) focus:text-(--ui-btn-color-hover) focus:bg-(--ui-btn-background-hover) focus:border-(--ui-btn-border-color-hover) active:text-(--ui-btn-color-active) active:bg-(--ui-btn-background-active) active:border-(--ui-btn-border-color-active) disabled:bg-(--ui-btn-background) disabled:border-(--ui-btn-border-color) aria-disabled:bg-(--ui-btn-background) aria-disabled:border-(--ui-btn-border-color) focus-visible:outline-(--ui-btn-background) ring-(--ui-btn-background-hover) focus:outline-none focus-visible:ring-(--ui-btn-background-hover) h-(--ui-btn-height) text-(length:--ui-btn-font-size) leading-(--ui-btn-height) font-(--ui-btn-font-weight) --style-filled ui-btn-md normal-case --air w-(--ui-btn-height) rounded-full">
  <!--v-if-->
  <div class="w-full inline-flex items-center justify-center gap-(--ui-btn-icon-space) ps-[4px] pe-[4px]"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" aria-hidden="true" data-slot="icon" class="text-(--ui-btn-color) shrink-0 size-(--ui-btn-icon-size)">
      <path fill="currentColor" fill-rule="evenodd" d="M14.05 15.886a6.37 6.37 0 1 1 1.836-1.837l3.311 3.311a1 1 0 0 1 0 1.414l-.422.423a1 1 0 0 1-1.414 0zm1.058-5.328a4.55 4.55 0 1 1-9.098 0 4.55 4.55 0 0 1 9.099 0" clip-rule="evenodd"></path>
    </svg>
    <!--v-if-->
    <!--v-if-->
  </div>
</button>"
`;

exports[`ChatPromptSubmit > renders with status error correctly 1`] = `
"<button type="button" aria-label="Send" class="ui-btn font-[family-name:var(--ui-font-family-primary)] select-none cursor-pointer inline-flex items-center relative outline-transparent focus-visible:outline-2 focus-visible:outline-offset-2 disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-30 aria-disabled:opacity-30 transition duration-0 ease-linear border-(length:--ui-btn-border-width) text-(--ui-btn-color) bg-(--ui-btn-background) border-(--ui-btn-border-color) hover:text-(--ui-btn-color-hover) hover:bg-(--ui-btn-background-hover) hover:border-(--ui-btn-border-color-hover) focus:text-(--ui-btn-color-hover) focus:bg-(--ui-btn-background-hover) focus:border-(--ui-btn-border-color-hover) active:text-(--ui-btn-color-active) active:bg-(--ui-btn-background-active) active:border-(--ui-btn-border-color-active) disabled:bg-(--ui-btn-background) disabled:border-(--ui-btn-border-color) aria-disabled:bg-(--ui-btn-background) aria-disabled:border-(--ui-btn-border-color) focus-visible:outline-(--ui-btn-background) ring-(--ui-btn-background-hover) focus:outline-none focus-visible:ring-(--ui-btn-background-hover) h-(--ui-btn-height) text-(length:--ui-btn-font-size) leading-(--ui-btn-height) font-(--ui-btn-font-weight) --style-filled-alert ui-btn-md normal-case --air w-(--ui-btn-height) rounded-full">
  <!--v-if-->
  <div class="w-full inline-flex items-center justify-center gap-(--ui-btn-icon-space) ps-[4px] pe-[4px]"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" aria-hidden="true" data-slot="icon" class="text-(--ui-btn-color) shrink-0 size-(--ui-btn-icon-size)">
      <path fill="currentColor" d="M19.203 4.633a.7.7 0 1 0-1.4 0V6.59c-2.016-2.16-5.18-2.994-8.108-2.163-3.351.95-5.054 3.918-5.431 7.153a.7.7 0 0 0 1.39.162c.337-2.887 1.798-5.224 4.423-5.968 2.492-.707 5.146.037 6.782 1.859h-2.055a.7.7 0 1 0 0 1.4h3.699a.7.7 0 0 0 .55-.268l.013-.016a.7.7 0 0 0 .137-.417zM4.946 15.634a.7.7 0 0 1 .7-.7h3.7a.7.7 0 1 1 0 1.4H7.288c1.64 1.807 4.287 2.544 6.775 1.84 2.628-.744 4.09-3.079 4.427-5.961a.7.7 0 0 1 1.39.162c-.377 3.233-2.083 6.197-5.436 7.146-2.922.827-6.079.002-8.097-2.143v1.956a.7.7 0 1 1-1.4 0v-3.7"></path>
    </svg>
    <!--v-if-->
    <!--v-if-->
  </div>
</button>"
`;

exports[`ChatPromptSubmit > renders with status ready correctly 1`] = `
"<button type="submit" aria-label="Send" class="ui-btn font-[family-name:var(--ui-font-family-primary)] select-none cursor-pointer inline-flex items-center relative outline-transparent focus-visible:outline-2 focus-visible:outline-offset-2 disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-30 aria-disabled:opacity-30 transition duration-0 ease-linear border-(length:--ui-btn-border-width) text-(--ui-btn-color) bg-(--ui-btn-background) border-(--ui-btn-border-color) hover:text-(--ui-btn-color-hover) hover:bg-(--ui-btn-background-hover) hover:border-(--ui-btn-border-color-hover) focus:text-(--ui-btn-color-hover) focus:bg-(--ui-btn-background-hover) focus:border-(--ui-btn-border-color-hover) active:text-(--ui-btn-color-active) active:bg-(--ui-btn-background-active) active:border-(--ui-btn-border-color-active) disabled:bg-(--ui-btn-background) disabled:border-(--ui-btn-border-color) aria-disabled:bg-(--ui-btn-background) aria-disabled:border-(--ui-btn-border-color) focus-visible:outline-(--ui-btn-background) ring-(--ui-btn-background-hover) focus:outline-none focus-visible:ring-(--ui-btn-background-hover) h-(--ui-btn-height) text-(length:--ui-btn-font-size) leading-(--ui-btn-height) font-(--ui-btn-font-weight) --style-filled ui-btn-md normal-case --air w-(--ui-btn-height) rounded-full">
  <!--v-if-->
  <div class="w-full inline-flex items-center justify-center gap-(--ui-btn-icon-space) ps-[4px] pe-[4px]"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" aria-hidden="true" data-slot="icon" class="text-(--ui-btn-color) shrink-0 size-(--ui-btn-icon-size)">
      <path fill="currentColor" d="m7.986 11.103 6.697.618a.247.247 0 0 1-.002.492l-6.686.56a1 1 0 0 0-.903.834l-.732 4.418a1 1 0 0 0 1.448 1.05l11.56-6.016a1 1 0 0 0 .019-1.764L7.84 4.988a1 1 0 0 0-1.466 1.044l.716 4.242a1 1 0 0 0 .895.829"></path>
    </svg>
    <!--v-if-->
    <!--v-if-->
  </div>
</button>"
`;

exports[`ChatPromptSubmit > renders with status streaming correctly 1`] = `
"<button type="button" aria-label="Send" class="ui-btn font-[family-name:var(--ui-font-family-primary)] select-none cursor-pointer inline-flex items-center relative outline-transparent focus-visible:outline-2 focus-visible:outline-offset-2 disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-30 aria-disabled:opacity-30 transition duration-0 ease-linear border-(length:--ui-btn-border-width) text-(--ui-btn-color) bg-(--ui-btn-background) border-(--ui-btn-border-color) hover:text-(--ui-btn-color-hover) hover:bg-(--ui-btn-background-hover) hover:border-(--ui-btn-border-color-hover) focus:text-(--ui-btn-color-hover) focus:bg-(--ui-btn-background-hover) focus:border-(--ui-btn-border-color-hover) active:text-(--ui-btn-color-active) active:bg-(--ui-btn-background-active) active:border-(--ui-btn-border-color-active) disabled:bg-(--ui-btn-background) disabled:border-(--ui-btn-border-color) aria-disabled:bg-(--ui-btn-background) aria-disabled:border-(--ui-btn-border-color) focus-visible:outline-(--ui-btn-background) ring-(--ui-btn-background-hover) focus:outline-none focus-visible:ring-(--ui-btn-background-hover) h-(--ui-btn-height) text-(length:--ui-btn-font-size) leading-(--ui-btn-height) font-(--ui-btn-font-weight) --style-outline-accent-2 ui-btn-md normal-case --air w-(--ui-btn-height) rounded-full">
  <!--v-if-->
  <div class="w-full inline-flex items-center justify-center gap-(--ui-btn-icon-space) ps-[4px] pe-[4px]"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" aria-hidden="true" data-slot="icon" class="text-(--ui-btn-color) shrink-0 size-(--ui-btn-icon-size)">
      <path fill="currentColor" fill-rule="evenodd" d="M7.5 5.5a2 2 0 0 0-2 2v9.002a2 2 0 0 0 2 2h9.002a2 2 0 0 0 2-2V7.5a2 2 0 0 0-2-2zm9.002 1.4H7.5a.6.6 0 0 0-.6.6v9.002a.6.6 0 0 0 .6.6h9.002a.6.6 0 0 0 .6-.6V7.5a.6.6 0 0 0-.6-.6" clip-rule="evenodd"></path>
    </svg>
    <!--v-if-->
    <!--v-if-->
  </div>
</button>"
`;

exports[`ChatPromptSubmit > renders with status submitted correctly 1`] = `
"<button type="button" aria-label="Send" class="ui-btn font-[family-name:var(--ui-font-family-primary)] select-none cursor-pointer inline-flex items-center relative outline-transparent focus-visible:outline-2 focus-visible:outline-offset-2 disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-30 aria-disabled:opacity-30 transition duration-0 ease-linear border-(length:--ui-btn-border-width) text-(--ui-btn-color) bg-(--ui-btn-background) border-(--ui-btn-border-color) hover:text-(--ui-btn-color-hover) hover:bg-(--ui-btn-background-hover) hover:border-(--ui-btn-border-color-hover) focus:text-(--ui-btn-color-hover) focus:bg-(--ui-btn-background-hover) focus:border-(--ui-btn-border-color-hover) active:text-(--ui-btn-color-active) active:bg-(--ui-btn-background-active) active:border-(--ui-btn-border-color-active) disabled:bg-(--ui-btn-background) disabled:border-(--ui-btn-border-color) aria-disabled:bg-(--ui-btn-background) aria-disabled:border-(--ui-btn-border-color) focus-visible:outline-(--ui-btn-background) ring-(--ui-btn-background-hover) focus:outline-none focus-visible:ring-(--ui-btn-background-hover) h-(--ui-btn-height) text-(length:--ui-btn-font-size) leading-(--ui-btn-height) font-(--ui-btn-font-weight) --style-outline-no-accent ui-btn-md normal-case --air w-(--ui-btn-height) rounded-full">
  <!--v-if-->
  <div class="w-full inline-flex items-center justify-center gap-(--ui-btn-icon-space) ps-[4px] pe-[4px]"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" aria-hidden="true" data-slot="icon" class="text-(--ui-btn-color) shrink-0 size-(--ui-btn-icon-size)">
      <path fill="currentColor" fill-rule="evenodd" d="M7.5 5.5a2 2 0 0 0-2 2v9.002a2 2 0 0 0 2 2h9.002a2 2 0 0 0 2-2V7.5a2 2 0 0 0-2-2zm9.002 1.4H7.5a.6.6 0 0 0-.6.6v9.002a.6.6 0 0 0 .6.6h9.002a.6.6 0 0 0 .6-.6V7.5a.6.6 0 0 0-.6-.6" clip-rule="evenodd"></path>
    </svg>
    <!--v-if-->
    <!--v-if-->
  </div>
</button>"
`;
